src/AWS.Deploy.CLI/Commands/TypeHints/VPCConnectorCommand.cs

```diff
@@ -0,0 +1,139 @@
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Amazon.AppRunner.Model;
+using Amazon.EC2.Model;
+using AWS.Deploy.CLI.TypeHintResponses;
+using AWS.Deploy.Common;
+using AWS.Deploy.Common.Recipes;
+using AWS.Deploy.Common.TypeHintData;
+using AWS.Deploy.Orchestration.Data;
+
+namespace AWS.Deploy.CLI.Commands.TypeHints
+{
+    public class VPCConnectorCommand : ITypeHintCommand
+    {
+        private readonly IAWSResourceQueryer _awsResourceQueryer;
+        private readonly IConsoleUtilities _consoleUtilities;
+        private readonly IToolInteractiveService _toolInteractiveService;
+
+        public VPCConnectorCommand(IAWSResourceQueryer awsResourceQueryer, IConsoleUtilities consoleUtilities, IToolInteractiveService toolInteractiveService)
+        {
+            _awsResourceQueryer = awsResourceQueryer;
+            _consoleUtilities = consoleUtilities;
+            _toolInteractiveService = toolInteractiveService;
+        }
+
+        private async Task<List<VpcConnector>> GetData()
+        {
+            return await _awsResourceQueryer.DescribeAppRunnerVpcConnectors();
+        }
+
+        public async Task<List<TypeHintResource>?> GetResources(Recommendation recommendation, OptionSettingItem optionSetting)
+        {
+            var vpcConnectors = await GetData();
+            return vpcConnectors.Select(vpcConnector => new TypeHintResource(vpcConnector.VpcConnectorArn, vpcConnector.VpcConnectorName)).ToList();
+        }
+
+        public async Task<object> Execute(Recommendation recommendation, OptionSettingItem optionSetting)
+        {
+            _toolInteractiveService.WriteLine();
+            var createNewOptionSetting = optionSetting.ChildOptionSettings.First(x => x.Id.Equals("CreateNew"));
+            var createNew = recommendation.GetOptionSettingValue<string>(createNewOptionSetting) ?? "false";
+            var createNewVPCConnectorAnswer = _consoleUtilities.AskYesNoQuestion("Do you want to create a new VPC Connector?", createNew);
+            var createNewVPCConnector = createNewVPCConnectorAnswer == YesNo.Yes;
+
+            _toolInteractiveService.WriteLine();
+            if (createNewVPCConnector)
+            {
+                _toolInteractiveService.WriteLine("In order to create a new VPC Connector, you need to select 1 or more Subnets as well as 1 or more Security groups.");
+                _toolInteractiveService.WriteLine();
+
+                var userInputConfigurationVPCs = new UserInputConfiguration<Vpc>(
+                    vpc => vpc.VpcId,
+                    vpc => vpc.VpcId,
+                    vpc => vpc.IsDefault)
+                {
+                    CanBeEmpty = false,
+                    CreateNew = false
+                };
+                var availableVpcs = await _awsResourceQueryer.GetListOfVpcs();
+                var vpc = _consoleUtilities.AskUserToChooseOrCreateNew<Vpc>(availableVpcs, "Select a VPC:", userInputConfigurationVPCs);
```

Comment:
If the user is going back through and editing the settings the default VPC should be the VPC last used. As I understand the logic now if I deployed to a non-default VPC then went back through this typehint it would default me back to the default VPC.

Reply:
I added the VPC ID as a setting so that it persists.